client/components/RelatedProducts/RelatedProducts.jsx

```diff
@@ -1,23 +1,56 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import RelatedProductsList from './Related/RelatedProductsList.jsx';
 import CustomOutfitList from './Outfit/CustomOutfitList.jsx';
 import Grid from '@material-ui/core/Grid';
-import products from './../DummyData/ProductDummyData';
+import axios from 'axios';
+import config from '../../../config.js'
 
+axios.defaults.headers.common['Authorization'] = config.GITHUB_TOKEN;
 
-const RelatedProducts = (id) => {
+const RelatedProducts = ({productId}) => {
 
-  //get request for related product ids based on id of current product
-  //then get request for each product to product_id (for name, category, default_price) AND get request to id/styles (for sale_price and image) as well??
-  // id = 24156;
-  let allProducts = products.products;
-  let relatedProducts = products.relatedProducts;
-  let relatedWithNameCatPrice = allProducts.filter(product => relatedProducts.includes(product.id));
+  const [relatedProductsData, setRelatedProductsData] = useState([]);
+
+  async function getRelatedIds(productId) {
+    const relatedIdsResponse = await axios.get(`https://app-hrsei-api.herokuapp.com/api/fec2/hratx/products/${productId}/related`);
+    return relatedIdsResponse.data;
+  }
+
+  useEffect(() => {
+    getRelatedIds(productId)
+    .then(relatedIdsResult => {
+
+      const productsIdGetReq = relatedIdsResult.map(id => {
+        return axios.get(`https://app-hrsei-api.herokuapp.com/api/fec2/hratx/products/${id}`);
+      })
+
+      const stylesGetReq = relatedIdsResult.map(id => {
+        return axios.get(`https://app-hrsei-api.herokuapp.com/api/fec2/hratx/products/${id}/styles`);
+      })
+
+      return Promise.all([...productsIdGetReq, ...stylesGetReq]);
+    })
+    .then(bothResponses => bothResponses.map(response => response.data))
+    .then(bothResponsesData => {
+
+      const midIndex = bothResponsesData.length / 2
+      const productsIdResult = bothResponsesData.slice(0, midIndex);
+      const stylesResult = bothResponsesData.slice(midIndex);
+      const combinedResults = [];
+
+      for (let i = 0; i < productsIdResult.length; i++) {
+        let temp = {...productsIdResult[i], ...stylesResult[i]};
```

Comment:
Definitely a clever solution! My thought is that maybe you could have done ```Promise.all``` twice to avoid having to split your array later on. Similarly, you could store the data in two pieces of state to avoid an ambiguous ```relatedProductsData``` variable. So much of what we put in variables is 'data' it can be a bit redundant to put it in a variable name, although I'm definitely guilty of it myself.  ```products``` and ```styles``` would give a reader a clearer picture of what it holds or even something like ```productsWithStyles``` if they're merged like this.

Reply:
So I did actually try both of those methods at one point, and both kept resulting in race conditions between the two.  Although I very well could have been implementing them incorrectly as well lol.